packages/core/src/ui/UIRenderer.ts

```diff
@@ -0,0 +1,193 @@
+import { Matrix, Plane, Ray, Vector3, Vector4 } from "@galacean/engine-math";
+import { DependentMode, dependentComponents } from "../ComponentsDependencies";
+import { Entity } from "../Entity";
+import { PrimitiveChunkManager } from "../RenderPipeline/PrimitiveChunkManager";
+import { RenderContext } from "../RenderPipeline/RenderContext";
+import { SubPrimitiveChunk } from "../RenderPipeline/SubPrimitiveChunk";
+import { Renderer } from "../Renderer";
+import { TransformModifyFlags } from "../Transform";
+import { ignoreClone } from "../clone/CloneManager";
+import { ComponentType } from "../enums/ComponentType";
+import { HitResult } from "../physics";
+import { ShaderProperty } from "../shader";
+import { ShaderMacroCollection } from "../shader/ShaderMacroCollection";
+import { CanvasGroup } from "./CanvasGroup";
+import { UICanvas } from "./UICanvas";
+import { UITransform } from "./UITransform";
+
+@dependentComponents(UITransform, DependentMode.AutoAdd)
+export class UIRenderer extends Renderer {
+  /** @internal */
+  static _tempVec30: Vector3 = new Vector3();
+  /** @internal */
+  static _tempVec31: Vector3 = new Vector3();
+  /** @internal */
+  static _tempMat: Matrix = new Matrix();
+  /** @internal */
+  static _tempPlane: Plane = new Plane();
+  /** @internal */
+  static _textureProperty: ShaderProperty = ShaderProperty.getByName("renderer_UITexture");
+  /** @internal */
+  @ignoreClone
+  _canvas: UICanvas;
+  /** @internal */
+  @ignoreClone
+  _group: CanvasGroup;
+  /** @internal */
+  @ignoreClone
+  _subChunk: SubPrimitiveChunk;
+
+  protected _alpha: number = 1;
+  protected _rayCastAble: boolean = true;
+  protected _rayCastPadding: Vector4 = new Vector4(0, 0, 0, 0);
```

Comment:
The type Vector4 in protected _rayCastPadding: Vector4 = new Vector4(0, 0, 0, 0);; can be omitted.